Bottom: 03147211e4c277726edc0f1afb8f0cde7fe7b6f2
Top:    73b8da5734e233017c6bb36d1dafd7d9997af036
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-06-26 13:43:05 -0500

Updates for ansible 2.4 (from 2.0)

Mainly syntax updates, also a few module name changes.


---
diff --git a/roles/bacula-catalog/tasks/main.yml b/roles/bacula-catalog/tasks/main.yml
index 03aacfa6..2d22328a 100644
--- a/roles/bacula-catalog/tasks/main.yml
+++ b/roles/bacula-catalog/tasks/main.yml
@@ -2,5 +2,5 @@
 - name: Check number of files in catalog
   command: /usr/bin/mysql bacula -B -e "select COUNT(*) From File;"
   register: filecount
-- include: restore-catalog-with-bsr.yml
+- include_tasks: restore-catalog-with-bsr.yml
   when: filecount.stdout_lines[1] == "0"
diff --git a/roles/bacula-director/tasks/main.yml b/roles/bacula-director/tasks/main.yml
index ca4799e6..06f130d8 100644
--- a/roles/bacula-director/tasks/main.yml
+++ b/roles/bacula-director/tasks/main.yml
@@ -5,22 +5,35 @@
     - bacula_monitor_password is defined
     - mariadb_bacula_password is defined
 - name: Install/setup Bacula Director
-  yum: name={{ item }} state=present
+  yum:
+    name: '{{ item }}'
+    state: present
   with_items:
     - bacula-director
 - name: Assemble dir for backup scripts
-  file: path=/usr/libexec/bacula/backup-scripts state=directory
+  file:
+    path: /usr/libexec/bacula/backup-scripts
+    state: directory
 - name: before_backup script part
-  template: src=before_backup.j2 dest=/usr/libexec/bacula/backup-scripts/30.before_backup.vm.{{ bacula_director_backup_vms }}
+  template:
+    src: 'before_backup.j2'
+    dest: '/usr/libexec/bacula/backup-scripts/30.before_backup.vm.{{ bacula_director_backup_vms }}'
 - name: after_backup script part
-  template: src=after_backup.j2 dest=/usr/libexec/bacula/backup-scripts/30.after_backup.vm.{{ bacula_director_backup_vms }}
+  template:
+    src: 'after_backup.j2'
+    dest: '/usr/libexec/bacula/backup-scripts/30.after_backup.vm.{{ bacula_director_backup_vms }}'
 - name: Configure Bacula Director
-  template: src=bacula-dir.conf.j2 dest={{ bacula_director_config_dir }}/bacula-dir.conf
-          owner=bacula group=bacula mode=0640 backup=yes
+  template:
+    src: bacula-dir.conf.j2
+    dest: '{{ bacula_director_config_dir }}/bacula-dir.conf'
+    owner: bacula
+    group: bacula
+    mode: 0640
+    backup: yes
   register: bacula_director_config_status
 - name: Use libbaccats-mysql
   command: alternatives --set libbaccats.so /usr/lib64/libbaccats-mysql.so
-- include: bacula-user.yml
+- import_tasks: bacula-user.yml
 - name: libvirt polkit
   copy: src=50-bacula-libvirt-access.pkla
         dest=/etc/polkit-1/localauthority/50-local.d/50-bacula-libvirt-access.pkla
@@ -29,20 +42,26 @@
   stat: path=/etc/selinux/targeted/modules/active/modules/baculavirt.pp
   register: baculavirt_pp
   when: bacula_director_backup_vms is defined
-- include: selinux-virt-policy.yml
+- import_tasks: selinux-virt-policy.yml
   when: bacula_director_backup_vms is defined and
         baculavirt_pp.stat.exists is defined and
         baculavirt_pp.stat.exists == false
 - name: Check if bacula mysql database exists
   command: mysql -u root -e "SHOW DATABASES;"
   register: databases
-- include: mariadb_setup.yml
+- import_tasks: mariadb_setup.yml
   when: databases.stdout.find('bacula') == -1
 - name: Test Bacula Director configuration
   command: /usr/sbin/bacula-dir -t -c {{ bacula_director_config_dir }}/bacula-dir.conf
 - name: Enable and restart Bacula Director
-  service: name=bacula-dir enabled=yes state=restarted
+  service:
+    name: bacula-dir
+    enabled: yes
+    state: restarted
   when: bacula_director_config_status.changed
 - name: Enable and start Bacula Director
-  service: name=bacula-dir enabled=yes state=started
+  service:
+    name: bacula-dir
+    enabled: yes
+    state: started
   when: not bacula_director_config_status.changed
diff --git a/roles/bacula-storage/tasks/main.yml b/roles/bacula-storage/tasks/main.yml
index 4cc7183a..c7074630 100644
--- a/roles/bacula-storage/tasks/main.yml
+++ b/roles/bacula-storage/tasks/main.yml
@@ -4,20 +4,34 @@
     - bacula_director_password is defined
     - bacula_monitor_password is defined
 - name: Install/setup Bacula Storage
-  yum: name={{ item }} state=present
+  yum:
+    name: '{{ item }}'
+    state: present
   with_items:
     - bacula-storage
 - name: Configure Bacula Storage
-  template: src=bacula-sd.conf.j2 dest={{ bacula_storage_config_dir }}/bacula-sd.conf
-          owner=bacula group=bacula mode=0640 backup=yes
+  template:
+    src: bacula-sd.conf.j2
+    dest: '{{ bacula_storage_config_dir }}/bacula-sd.conf'
+    owner: bacula
+    group: bacula
+    mode: 0640
+    backup: yes
+
 - name: Check if baculatape selinux module exists
   stat: path=/etc/selinux/targeted/modules/active/modules/baculatape.pp
   register: baculatape_pp
-- include: selinux-tape-policy.yml
+- import_tasks: selinux-tape-policy.yml
   when: baculatape_pp.stat.exists is defined and baculatape_pp.stat.exists == false
 - name: Make sure bacula user is in the tape group
-  user: name=bacula groups=tape append=yes
+  user:
+    name: bacula
+    groups: tape
+    append: yes
 - name: Test Bacula Storage configuration
   command: /usr/sbin/bacula-sd -t -c {{ bacula_storage_config_dir }}/bacula-sd.conf
 - name: Enable and start Bacula Storage
-  service: name=bacula-sd enabled=yes state=started
+  service:
+    name: bacula-sd
+    enabled: yes
+    state: started
diff --git a/roles/bacula-storage/tasks/selinux-tape-policy.yml b/roles/bacula-storage/tasks/selinux-tape-policy.yml
index 7fb68000..141bbf8f 100644
--- a/roles/bacula-storage/tasks/selinux-tape-policy.yml
+++ b/roles/bacula-storage/tasks/selinux-tape-policy.yml
@@ -2,9 +2,12 @@
   command: mktemp -d /tmp/ansible-bacula-storage.XXXXXXXX
   register: bacula_storage_tmp_dir
 - name: Copy baculatape type enforcement file
-  copy: src=baculatape.te
-          dest={{ bacula_storage_tmp_dir.stdout }}/baculatape.te
-          owner=root group=root mode=0640
+  copy:
+    src: baculatape.te
+    dest: '{{ bacula_storage_tmp_dir.stdout }}/baculatape.te'
+    owner: root
+    group: root
+    mode: 0640
 - name: Run checkmodule on baculatape.te
   command: /usr/bin/checkmodule -M -m -o baculatape.mod baculatape.te
   args:
@@ -18,4 +21,6 @@
   args:
     chdir: "{{ bacula_storage_tmp_dir.stdout }}"
 - name: Delete temp directory
-  file: path={{ bacula_storage_tmp_dir.stdout }} state=absent
+  file:
+    path: '{{ bacula_storage_tmp_dir.stdout }} '
+    state: absent
diff --git a/roles/buildsystem/tasks/grab_files/ncd.yml b/roles/buildsystem/tasks/grab_files/ncd.yml
index 553785b9..001007e7 100644
--- a/roles/buildsystem/tasks/grab_files/ncd.yml
+++ b/roles/buildsystem/tasks/grab_files/ncd.yml
@@ -1,7 +1,7 @@
 ---
 - name: Check for local directory to store NCD SVN Export
   run_once: true
-  sudo: false
+  become: false
   local_action:
     module: file
     mode: 0775
@@ -10,7 +10,7 @@
 
 - name: Gathering needed NCD files locally
   run_once: true
-  sudo: false
+  become: false
   local_action:
     module: ncdrelease
     svn_username: "{{ username }}"
diff --git a/roles/buildsystem/tasks/grab_files/web.yml b/roles/buildsystem/tasks/grab_files/web.yml
index 2fbc5962..88316e8a 100644
--- a/roles/buildsystem/tasks/grab_files/web.yml
+++ b/roles/buildsystem/tasks/grab_files/web.yml
@@ -1,7 +1,7 @@
 ---
 - name: Check for local directory to store the web site code exists
   run_once: true
-  sudo: false
+  become: false
   local_action:
     module: file
     mode: 0775
@@ -10,7 +10,7 @@
 
 - name: Downloading web site code
   run_once: true
-  sudo: false
+  become: false
   local_action:
     module: git
     repo: "{{ web_site_repo_url }}"
@@ -18,21 +18,21 @@
     force: yes
 
 - name: Set datebase name in website code
-  sudo: false
+  become: false
   local_action: replace
     dest="{{ working_directory }}/{{ web_site_name }}/settings.py"
     regexp="(.*'NAME':\s*').*('.*)"
     replace='\1{{ database_name }}\2'
 
 - name: Set database user in website code
-  sudo: false
+  become: false
   local_action: replace
     dest="{{ working_directory }}/{{ web_site_name }}/settings.py"
     regexp="(.*'USER':\s*').*('.*)"
     replace='\1{{ database_user }}\2'
 
 - name: Set database user password in website code
-  sudo: false
+  become: false
   local_action: replace
     dest="{{ working_directory }}/{{ web_site_name }}/settings.py"
     regexp="(.*'PASSWORD':\s*).*"
diff --git a/roles/buildsystem/tasks/main.yml b/roles/buildsystem/tasks/main.yml
index 0316be2b..bff70495 100644
--- a/roles/buildsystem/tasks/main.yml
+++ b/roles/buildsystem/tasks/main.yml
@@ -6,14 +6,15 @@
     - svn_username is defined
     - svn_password is defined
 
-- include: grab_files/ncd.yml
-  username: "{{ svn_username }}"
-  password: "{{ svn_password }}"
-  repo: "{{ ncd_repo }}"
-  working_directory: "{{ local_working_directory }}"
+- import_tasks: grab_files/ncd.yml
+  vars:
+    username: "{{ svn_username }}"
+    password: "{{ svn_password }}"
+    repo: "{{ ncd_repo }}"
+    working_directory: "{{ local_working_directory }}"
 
 
-- include: grab_files/web.yml
+- import_tasks: grab_files/web.yml
   vars:
     working_directory: "{{ local_working_directory }}"
     web_site_name: BuildSystem
@@ -24,7 +25,7 @@
     server_directory: "{{ buildsystem_directory }}"
 
 
-- include: create_db_user.yml
+- import_tasks: create_db_user.yml
   vars:
     db_host:                 "test_station_mysql_server"
     db_admin_user:           "root"
diff --git a/roles/docker-build-base-image/tasks/main.yml b/roles/docker-build-base-image/tasks/main.yml
index 0843c142..b984d740 100644
--- a/roles/docker-build-base-image/tasks/main.yml
+++ b/roles/docker-build-base-image/tasks/main.yml
@@ -13,11 +13,11 @@
     recursive: yes
     update: yes
     version: '{{ item.value.commit_id }}'
-  with_dict: docker_build_base_images
+  with_dict: "{{ docker_build_base_images }}"
 
 - name: build image
   docker_build:
     image_name: '{{ item.value.image_name }}'
     image_tag: '{{ item.value.image_tag }}'
     dockerfile_dir: '{{ docker_projects_dir }}/base-images/{{ item.key }}/{{ item.value.relative_dir }}'
-  with_dict: docker_build_base_images
+  with_dict: "{{ docker_build_base_images }}"
diff --git a/roles/docker-buildsystem/defaults/main.yml b/roles/docker-buildsystem/defaults/main.yml
index 571e7b25..cdbad40e 100644
--- a/roles/docker-buildsystem/defaults/main.yml
+++ b/roles/docker-buildsystem/defaults/main.yml
@@ -20,7 +20,7 @@ name_prefix: ""
 image_build_state_data: present
 container_start_state_data: present
 image_build_state_nondata: present
-container_start_state_nondata: reloaded
+container_start_state_nondata: started
 
 build_system_docker_repos:
   TestStation_http_data_buildsystem:
diff --git a/roles/docker-buildsystem/tasks/docker_containers/datacontainer_buildsystem.yml b/roles/docker-buildsystem/tasks/docker_containers/datacontainer_buildsystem.yml
index ae22977e..34167b2f 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/datacontainer_buildsystem.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/datacontainer_buildsystem.yml
@@ -9,7 +9,7 @@
     state: "{{ image_build_state_data }}"
 
 - name: Start docker container - Test Station build system data
-  docker: 
+  docker_container: 
     detach: True
     image: "{{ name_prefix }}teststation_http_data_buildsystem_image"
     name: "{{ name_prefix }}TestStation_http_data_buildsystem"
diff --git a/roles/docker-buildsystem/tasks/docker_containers/datacontainer_database.yml b/roles/docker-buildsystem/tasks/docker_containers/datacontainer_database.yml
index 31f803b7..3f6150c4 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/datacontainer_database.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/datacontainer_database.yml
@@ -9,7 +9,7 @@
     state: "{{ image_build_state_data }}"
 
 - name: Start docker container - Test Station database data
-  docker: 
+  docker_container: 
     detach: True
     image: "{{ name_prefix }}teststation_database_data_image"
     name: "{{ name_prefix }}TestStation_database_data"
diff --git a/roles/docker-buildsystem/tasks/docker_containers/datacontainer_ipkg.yml b/roles/docker-buildsystem/tasks/docker_containers/datacontainer_ipkg.yml
index 49c59f5e..1c867665 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/datacontainer_ipkg.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/datacontainer_ipkg.yml
@@ -9,7 +9,7 @@
     state: "{{ image_build_state_data }}"
 
 - name: Start docker container - Test Station ipkg data
-  docker: 
+  docker_container: 
     detach: True
     image: "{{ name_prefix }}teststation_http_data_ipkg_image"
     name: "{{ name_prefix }}TestStation_http_data_ipkg"
diff --git a/roles/docker-buildsystem/tasks/docker_containers/datacontainer_iso.yml b/roles/docker-buildsystem/tasks/docker_containers/datacontainer_iso.yml
index 9a330824..27421a72 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/datacontainer_iso.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/datacontainer_iso.yml
@@ -9,7 +9,7 @@
     state: "{{ image_build_state_data }}"
 
 - name: Start docker container - Test Station ncd iso data
-  docker: 
+  docker_container: 
     detach: True
     image: "{{ name_prefix }}teststation_http_data_ncdiso_image"
     name: "{{ name_prefix }}TestStation_http_data_ncdiso"
diff --git a/roles/docker-buildsystem/tasks/docker_containers/datacontainer_ncdrelease.yml b/roles/docker-buildsystem/tasks/docker_containers/datacontainer_ncdrelease.yml
index 6cd9f82b..0c950bab 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/datacontainer_ncdrelease.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/datacontainer_ncdrelease.yml
@@ -9,7 +9,7 @@
     state: "{{ image_build_state_data }}"
 
 - name: Start docker container - Test Station ncd release data
-  docker: 
+  docker_container: 
     detach: True
     image: "{{ name_prefix }}teststation_http_data_ncdrelease_image"
     name: "{{ name_prefix }}TestStation_http_data_ncdrelease"
diff --git a/roles/docker-buildsystem/tasks/docker_containers/datacontainer_testclient.yml b/roles/docker-buildsystem/tasks/docker_containers/datacontainer_testclient.yml
index 280f26f7..47db0a2f 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/datacontainer_testclient.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/datacontainer_testclient.yml
@@ -9,7 +9,7 @@
     state: "{{ image_build_state_data }}"
 
 - name: Start docker container - Test Station  test client data
-  docker: 
+  docker_container: 
     detach: True
     image: "{{ name_prefix }}teststation_http_data_testclient_image"
     name: "{{ name_prefix }}TestStation_http_data_testclient"
diff --git a/roles/docker-buildsystem/tasks/docker_containers/datacontainer_tftp.yml b/roles/docker-buildsystem/tasks/docker_containers/datacontainer_tftp.yml
index 755bbf8b..8b296f1b 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/datacontainer_tftp.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/datacontainer_tftp.yml
@@ -9,7 +9,7 @@
     state: "{{ image_build_state_data }}"
 
 - name: Start docker container - Test Station tftp data
-  docker: 
+  docker_container: 
     detach: True
     image: "{{ name_prefix }}teststation_tftp_data_image"
     name: "{{ name_prefix }}TestStation_tftp_data"
diff --git a/roles/docker-buildsystem/tasks/docker_containers/server_database.yml b/roles/docker-buildsystem/tasks/docker_containers/server_database.yml
index 1e990c6a..1ab11330 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/server_database.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/server_database.yml
@@ -21,7 +21,7 @@
     state: "{{ image_build_state_nondata }}"
 
 - name: Start docker container - Test Station database server
-  docker:
+  docker_container:
     detach: True
     image: "{{ name_prefix }}teststation_database_server_image"
     name: "{{ name_prefix }}TestStation_database_server"
diff --git a/roles/docker-buildsystem/tasks/docker_containers/server_http.yml b/roles/docker-buildsystem/tasks/docker_containers/server_http.yml
index ea1bff7e..883d3f4b 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/server_http.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/server_http.yml
@@ -9,7 +9,7 @@
     state: "{{ image_build_state_nondata }}"
 
 - name: Start docker container - Test Station http server
-  docker:
+  docker_container:
     detach: True
     image: "{{ name_prefix }}teststation_http_image"
     links: "{{ name_prefix }}TestStation_database_server:test_station_mysql_server"
diff --git a/roles/docker-buildsystem/tasks/docker_containers/server_ssh.yml b/roles/docker-buildsystem/tasks/docker_containers/server_ssh.yml
index 223b529a..1e3333b4 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/server_ssh.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/server_ssh.yml
@@ -15,7 +15,7 @@
     state: "{{ image_build_state_nondata }}"
 
 - name: Start docker container - Test Station ssh server
-  docker:
+  docker_container:
     detach: True
     image: "{{ name_prefix }}teststation_ssh_image"
     links: "{{ name_prefix }}TestStation_database_server:test_station_mysql_server"
@@ -31,4 +31,3 @@
       - "{{ name_prefix }}TestStation_http_data_ncdrelease"
       - "{{ name_prefix }}TestStation_http_data_testclient"
       - "{{ name_prefix }}TestStation_tftp_data"
-
diff --git a/roles/docker-buildsystem/tasks/docker_containers/server_ssh_prod.yml b/roles/docker-buildsystem/tasks/docker_containers/server_ssh_prod.yml
index 3b718a6d..6f95f9e2 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/server_ssh_prod.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/server_ssh_prod.yml
@@ -27,7 +27,7 @@
     state: "{{ image_build_state_nondata }}"
 
 - name: Start docker container - Test Station production ssh server temp.
-  docker:
+  docker_container:
     detach: True
     image: "{{ name_prefix }}teststation_ssh_server_manualtest_image"
     name: "{{ name_prefix }}TestStation_ssh_server_manualtest"
diff --git a/roles/docker-buildsystem/tasks/docker_containers/server_tftp.yml b/roles/docker-buildsystem/tasks/docker_containers/server_tftp.yml
index 27609613..a6498fcf 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/server_tftp.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/server_tftp.yml
@@ -9,7 +9,7 @@
     state: "{{ image_build_state_nondata }}"
 
 - name: Start docker container - Test Station tftp server
-  docker:
+  docker_container:
     detach: True
     image: "{{ name_prefix }}teststation_tftp_image"
     name: "{{ name_prefix }}TestStation_tftp_server"
diff --git a/roles/docker-buildsystem/tasks/docker_containers/service_cron.yml b/roles/docker-buildsystem/tasks/docker_containers/service_cron.yml
index 77372198..94246679 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/service_cron.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/service_cron.yml
@@ -9,7 +9,7 @@
     state: "{{ image_build_state_nondata }}"
 
 - name: Start docker container - Test Station cron service
-  docker:
+  docker_container:
     detach: True
     image: "{{ name_prefix }}teststation_cron_image"
     name: "{{ name_prefix }}TestStation_cron_service"
diff --git a/roles/docker-buildsystem/tasks/main.yml b/roles/docker-buildsystem/tasks/main.yml
index 3c8b7d13..b62739ea 100644
--- a/roles/docker-buildsystem/tasks/main.yml
+++ b/roles/docker-buildsystem/tasks/main.yml
@@ -37,22 +37,22 @@
   with_dict: "{{ build_system_docker_repos }}"
 
 
-- include: docker_containers/datacontainer_buildsystem.yml
-- include: docker_containers/datacontainer_database.yml
-#- include: docker_containers/datacontainer_ipkg.yml
-- include: docker_containers/datacontainer_iso.yml
-- include: docker_containers/datacontainer_ncdrelease.yml
-- include: docker_containers/datacontainer_testclient.yml
-- include: docker_containers/datacontainer_tftp.yml
-
-- include: docker_containers/server_database.yml
-- include: docker_containers/service_cron.yml
-- include: docker_containers/server_tftp.yml
-- include: docker_containers/server_ssh.yml
-- include: docker_containers/server_http.yml
-- include: docker_containers/server_ssh_prod.yml
-
-- include: docker_containers/service_backup.yml
+- import_tasks: docker_containers/datacontainer_buildsystem.yml
+- import_tasks: docker_containers/datacontainer_database.yml
+#- import_tasks: docker_containers/datacontainer_ipkg.yml
+- import_tasks: docker_containers/datacontainer_iso.yml
+- import_tasks: docker_containers/datacontainer_ncdrelease.yml
+- import_tasks: docker_containers/datacontainer_testclient.yml
+- import_tasks: docker_containers/datacontainer_tftp.yml
+
+- import_tasks: docker_containers/server_database.yml
+- import_tasks: docker_containers/service_cron.yml
+- import_tasks: docker_containers/server_tftp.yml
+- import_tasks: docker_containers/server_ssh.yml
+- import_tasks: docker_containers/server_http.yml
+- import_tasks: docker_containers/server_ssh_prod.yml
+
+- import_tasks: docker_containers/service_backup.yml
 
 - name: Copy Build System default ssh key to user directroy
   fetch:
@@ -61,7 +61,7 @@
     flat: yes
 
 - name: Setting ssh key file permissions
-  sudo: false
+  become: false
   local_action:
     module: file
     path: ~/.ssh/id_rsa_buildsystem_default
@@ -131,5 +131,5 @@
 # *****************************************************************************
 # restore?
 
-- include: restore.yml
+- import_tasks: restore.yml
   when: st_buildsystem_restore.stat.exists == False
diff --git a/roles/docker-common/tasks/ip_addr.yml b/roles/docker-common/tasks/ip_addr.yml
index 885f8f39..6f8d9656 100644
--- a/roles/docker-common/tasks/ip_addr.yml
+++ b/roles/docker-common/tasks/ip_addr.yml
@@ -9,9 +9,10 @@
     owner: root
     group: root
     mode: 'u=rw,g=r,o=r'
-  when: ansible_os_family == "RedHat"
-  when: (item.value.bridge_num is not undefined) and (ansible_os_family == "RedHat")
-  with_dict: container_addr_map
+  when: 
+    - ansible_os_family == "RedHat"
+    - (item.value.bridge_num is not undefined) and (ansible_os_family == "RedHat")
+  with_dict: "{{ container_addr_map }}"
 
 - name: add container hostname to /etc/hosts
   lineinfile:
@@ -20,7 +21,7 @@
     line: "{{ item.value.ip_addr }}	{{ item.value.hostname }}"
     state: present
   when: item.value.bridge_num is not undefined
-  with_dict: container_addr_map
+  with_dict: "{{ container_addr_map }}"
 
 - name: enable ip forwarding
   lineinfile:
diff --git a/roles/docker-common/tasks/main.yml b/roles/docker-common/tasks/main.yml
index c812e958..85be0141 100644
--- a/roles/docker-common/tasks/main.yml
+++ b/roles/docker-common/tasks/main.yml
@@ -12,10 +12,10 @@
     state: restarted
     enabled: yes
 
-- include: ip_addr.yml
-- include: packages.yml
-- include: sshd_settings.yml
-- include: mariadb_settings.yml
+- import_tasks: ip_addr.yml
+- import_tasks: packages.yml
+- import_tasks: sshd_settings.yml
+- import_tasks: mariadb_settings.yml
 
 - name: restores base dir
   file:
@@ -49,7 +49,7 @@
     clone: yes
     recursive: yes
     update: yes
-  with_items: docker_projects
+  with_items: "{{ docker_projects }}"
   tags:
     - docker_repositories
 
@@ -59,7 +59,7 @@
   register: docker_bacula_pp
   when: ansible_os_family == "RedHat"
 
-- include: selinux-backula-policy.yml
+- include_tasks: selinux-backula-policy.yml
   when: (docker_bacula_pp.stat.exists is defined) and (docker_bacula_pp.stat.exists == false)
 
 # *****************************************************************************
diff --git a/roles/docker-common/tasks/sshd_settings.yml b/roles/docker-common/tasks/sshd_settings.yml
index 31d37960..933c7f23 100644
--- a/roles/docker-common/tasks/sshd_settings.yml
+++ b/roles/docker-common/tasks/sshd_settings.yml
@@ -7,12 +7,14 @@
     regexp: '#ListenAddress 0.0.0.0'
     replace: 'ListenAddress {{ ansible_br0["ipv4"]["address"] }}'
     backup: yes
-  when: ansible_os_family == "RedHat"
-  when: ansible_br0 is defined and ansible_br0.ipv4 is defined and ansible_br0.ipv4.address is defined
+  when: 
+    - ansible_os_family == "RedHat"
+    - ansible_br0 is defined and ansible_br0.ipv4 is defined and ansible_br0.ipv4.address is defined
 
 - name: Restart sshd server
   service:
     name: sshd
     state: restarted
-  when: ansible_os_family == "RedHat"
-  when: ansible_br0 is defined and ansible_br0.ipv4 is defined and ansible_br0.ipv4.address is defined
+  when: 
+    - ansible_os_family == "RedHat"
+    - ansible_br0 is defined and ansible_br0.ipv4 is defined and ansible_br0.ipv4.address is defined
diff --git a/roles/docker-exim4/tasks/main.yml b/roles/docker-exim4/tasks/main.yml
index 56072569..0c155b43 100644
--- a/roles/docker-exim4/tasks/main.yml
+++ b/roles/docker-exim4/tasks/main.yml
@@ -2,7 +2,7 @@
 # file: roles/docker-exim4/tasks/main.yaml
 
 # set the selinux policy to enable log-rotation of the exim4 logs
-- include: selinux-docker-log_rotate.yml
+- import_tasks: selinux-docker-log_rotate.yml
 
 # *****************************************************************************
 # Setup the directory where the backup and restore is to take place
@@ -102,7 +102,7 @@
 # *****************************************************************************
 # restore?
 
-- include: restore.yml
+- include_tasks: restore.yml
   when: st_exim4_restore.stat.exists == False
 
 # *****************************************************************************
diff --git a/roles/docker-gitlab/tasks/main.yml b/roles/docker-gitlab/tasks/main.yml
index b00114a4..8404097a 100644
--- a/roles/docker-gitlab/tasks/main.yml
+++ b/roles/docker-gitlab/tasks/main.yml
@@ -103,7 +103,7 @@
     data_volume_container_name: '{{ gitlab_db_dv_name }}'
 
 - name: stop prev container (postgres)
-  docker:
+  docker_container:
     image: '{{ gitlab_db_image_name }}'
     name: '{{ gitlab_db_container_name }}'
     state: stopped
@@ -148,5 +148,5 @@
 # *****************************************************************************
 # restore?
 
-- include: restore.yml
+- include_tasks: restore.yml
   when: st_gitlab_restore.stat.exists == False
diff --git a/roles/docker-mantisbt/tasks/main.yml b/roles/docker-mantisbt/tasks/main.yml
index c547506d..1f1e817c 100644
--- a/roles/docker-mantisbt/tasks/main.yml
+++ b/roles/docker-mantisbt/tasks/main.yml
@@ -111,8 +111,8 @@
     data_volume_container_name: '{{ mantisbt_db_dv_name }}'
 
 - name: stop prev container (mysql)
-  docker:
-    image: '{{ mantisbt_db_image_name }}'
+  docker_container:
+    image: '{{ mantisbt_db_image_name }}:{{ docker_image_tag }}'
     name: '{{ mantisbt_db_container_name }}'
     state: stopped
 
@@ -157,5 +157,5 @@
 # *****************************************************************************
 # restore?
 
-- include: restore.yml
+- include_tasks: restore.yml
   when: st_mantisbt_restore.stat.exists == False
diff --git a/roles/docker-mediawiki/tasks/main.yml b/roles/docker-mediawiki/tasks/main.yml
index d8e9f7ff..dabe61e1 100644
--- a/roles/docker-mediawiki/tasks/main.yml
+++ b/roles/docker-mediawiki/tasks/main.yml
@@ -93,7 +93,7 @@
     data_volume_container_name: '{{ wiki_db_dv_name }}'
 
 - name: stop prev container (mysql)
-  docker:
+  docker_container:
     image: '{{ wiki_db_image_name }}'
     name: '{{ wiki_db_container_name }}'
     state: stopped
@@ -125,5 +125,5 @@
 # *****************************************************************************
 # restore?
 
-- include: restore.yml
+- include_tasks: restore.yml
   when: st_mediawiki_restore.stat.exists == False
diff --git a/roles/docker-openldap/tasks/main.yml b/roles/docker-openldap/tasks/main.yml
index f425302c..51ceb46b 100644
--- a/roles/docker-openldap/tasks/main.yml
+++ b/roles/docker-openldap/tasks/main.yml
@@ -99,7 +99,7 @@
 # *****************************************************************************
 # restore?
 
-- include: restore.yml
+- include_tasks: restore.yml
   when: st_openldap_restore.stat.exists == False
 
 # *****************************************************************************
diff --git a/roles/docker-openssl/tasks/main.yml b/roles/docker-openssl/tasks/main.yml
index 589ddede..2c0f093b 100644
--- a/roles/docker-openssl/tasks/main.yml
+++ b/roles/docker-openssl/tasks/main.yml
@@ -79,5 +79,5 @@
 # *****************************************************************************
 # restore?
 
-- include: restore.yml
+- include_tasks: restore.yml
   when: st_openssl_restore.stat.exists == False
diff --git a/roles/docker-pull-images/tasks/main.yml b/roles/docker-pull-images/tasks/main.yml
index e0dc0295..fadb35ee 100644
--- a/roles/docker-pull-images/tasks/main.yml
+++ b/roles/docker-pull-images/tasks/main.yml
@@ -9,6 +9,6 @@
     image_name: '{{ item.value.image }}'
     pull_image_tag: '{{ item.value.tag }}'
     working_image_tag: '{{ docker_image_tag }}'
-  with_dict: repository_images
+  with_dict: "{{ repository_images }}"
   tags:
     - docker_pull
diff --git a/roles/docker-svn/tasks/main.yml b/roles/docker-svn/tasks/main.yml
index faee9e67..fffe49a5 100644
--- a/roles/docker-svn/tasks/main.yml
+++ b/roles/docker-svn/tasks/main.yml
@@ -80,5 +80,5 @@
 # *****************************************************************************
 # restore?
 
-- include: restore.yml
+- include_tasks: restore.yml
   when: st_svn_restore.stat.exists == False
diff --git a/roles/mariadb/tasks/main.yml b/roles/mariadb/tasks/main.yml
index dadffc4d..b27a0af7 100644
--- a/roles/mariadb/tasks/main.yml
+++ b/roles/mariadb/tasks/main.yml
@@ -3,20 +3,42 @@
     that:
     - mariadb_root_password is defined
 - name: Install MariaDB
-  yum: name={{ item }} state=present
+  yum:
+    name: '{{ item }}'
+    state: present
   with_items:
     - mariadb-server
     - mariadb
 - name: Config MariaDB server
-  template: src=ansible.cnf.j2 dest={{ mariadb_server_config_dir }}/ansible.cnf
-            owner=root group=root mode=0644 backup=yes
+  template:
+    src: ansible.cnf.j2
+    dest: '{{ mariadb_server_config_dir }}/ansible.cnf'
+    owner: root
+    group: root
+    mode: 0644
+    backup: yes
+- name: Set host mysqld bind-address
+  template:
+    src: mariadb-bind-address.j2
+    dest: '{{ mariadb_server_config_dir }}/mariadb-bind-address.cnf'
+    owner: root
+    group: root
+    mode: 0644
+    backup: yes
 - name: Enable and start MariaDB service
-  service: name=mariadb enabled=yes state=started
+  service:
+    name: mariadb
+    enabled: yes
+    state: started
 - name: Check previous installation
-  stat: path={{ mariadb_client_config }}
+  stat:
+    path: '{{ mariadb_client_config }}'
   register: cfg
-- include: initial_config.yml
+- include_tasks: initial_config.yml
   when: cfg.stat.exists is defined and cfg.stat.exists == false
 - name: Open the firewall
-  firewalld: service=mysql permanent=true state=enabled
+  firewalld:
+    service: mysql
+    permanent: true
+    state: enabled
   when: mariadb_open_firewall
